File: scripts/download-models.js
const fs = require('fs');
const https = require('https');
const path = require('path');

const MODELS_URL = "https://raw.githubusercontent.com/justadudewhohacks/face-api.js/master/weights/";
const TARGET_DIR = path.join(process.cwd(), "public", "models");

const modelsToDownload = [
    // SSD Mobilenet V1
    "ssd_mobilenetv1_model-weights_manifest.json",
    "ssd_mobilenetv1_model-shard1",
    "ssd_mobilenetv1_model-shard2",
    // Face Landmark 68
    "face_landmark_68_model-weights_manifest.json",
    "face_landmark_68_model-shard1",
    // Face Recognition
    "face_recognition_model-weights_manifest.json",
    "face_recognition_model-shard1",
];

if (!fs.existsSync(TARGET_DIR)) {
    fs.mkdirSync(TARGET_DIR, { recursive: true });
}

function download(url, dest) {
    return new Promise((resolve, reject) => {
        const file = fs.createWriteStream(dest);
        https.get(url, (response) => {
            if (response.statusCode !== 200) {
                reject(new Error(`Failed to download ${url}: HTTP ${response.statusCode}`));
                return;
            }
            response.pipe(file);
            file.on('finish', () => {
                file.close();
                console.log(`Successfully downloaded ${path.basename(dest)}`);
                resolve();
            });
        }).on('error', (err) => {
            fs.unlink(dest, () => {});
            reject(err);
        });
    });
}

async function start() {
    for (const model of modelsToDownload) {
        const url = MODELS_URL + model;
        const dest = path.join(TARGET_DIR, model);
        console.log(`Downloading ${model}...`);
        try {
            await download(url, dest);
        } catch (err) {
            console.error(`Error downloading ${model}: ${err.message}`);
        }
    }
    console.log("All downloads finished!");
}

start();

File: scripts/download-models.py
import os
import requests

MODELS_URL = "https://raw.githubusercontent.com/justadudewhohacks/face-api.js/master/weights/"
TARGET_DIR = "public/models"

models_to_download = [
    # SSD Mobilenet V1
    "ssd_mobilenet_v1_model-weights_manifest.json",
    "ssd_mobilenet_v1_model-shard1",
    "ssd_mobilenet_v1_model-shard2",
    # Face Landmark 68
    "face_landmark_68_model-weights_manifest.json",
    "face_landmark_68_model-shard1",
    # Face Recognition
    "face_recognition_model-weights_manifest.json",
    "face_recognition_model-shard1",
]

if not os.path.exists(TARGET_DIR):
    os.makedirs(TARGET_DIR)

for model in models_to_download:
    url = MODELS_URL + model
    target_path = os.path.join(TARGET_DIR, model)
    print(f"Downloading {model}...")
    try:
        response = requests.get(url)
        if response.status_code == 200:
            with open(target_path, "wb") as f:
                f.write(response.content)
            print(f"Successfully downloaded {model}")
        else:
            print(f"Failed to download {model}: HTTP {response.status_code}")
    except Exception as e:
        print(f"Error downloading {model}: {e}")
